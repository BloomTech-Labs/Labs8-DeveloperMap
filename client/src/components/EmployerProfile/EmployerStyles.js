import styled from 'styled-components';

export const Info = styled.div`

  max-width: 300px;
  width: 100%;
  padding: 5%;
  .com-name{
    font-size: 1.6rem;
    margin-bottom: 30px;
  }
  h3 {
    font-size: 1.4rem;
    margin-bottom: 10px;
  }
  h4{
    margin-bottom: 30px;
  }
`;


export const Styling = styled.div`
  display: flex;
  padding: 5%;
  justify-content: space-between;
  max-width: 750px;
  width: 100%;
`;

export const Posts = styled.div`
  width: 100%;
`;

export const PostContainer = styled.div`
  display: flex;
  justify-content: center;
  max-width: 400px;
  width: 100%;
`;