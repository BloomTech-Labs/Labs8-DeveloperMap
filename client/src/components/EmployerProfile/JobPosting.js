import React from 'react';
import styled from 'styled-components';
import Heart from '../../images/hollow-heart.png';
import FavHeart from '../../images/favorites-icon.png';

class JobPosting extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
  componentDidMount() {
    // Axios.get().then(response => {
    //   const favoritedList = response.data;
    //   if (favoritedList.includes(this.props.post.jobId)) {
    //     document.querySelector(`#${jobId}`).setAttribute('src', FavHeart);
    //   }
    // });
    const favoritedList = ['jb1'];
    if (favoritedList.includes(this.props.post.jobId)) {
      document
        .querySelector(`#${this.props.post.jobId}`)
        .setAttribute('src', FavHeart);
    }
  }

  passPostThenToggle = e => {
    this.props.favToggle(e, this.props.post);
  };
  render() {
    return (
      <Job>
        <div>
          <h2>{this.props.post.companyName}</h2>
          <h3>{this.props.post.date}</h3>
          <h4>{this.props.post.jobLink}</h4>
          <h4>{this.props.post.jobTitle}</h4>
          <h4>{this.props.post.location}</h4>
        </div>
        <div>
          <img
            id={this.props.post.jobId}
            alt="hollow heart"
            src={Heart}
            style={{ width: '50px' }}
            onClick={this.passPostThenToggle}
          />
        </div>
      </Job>
    );
  }
}

const Job = styled.div`
  box-shadow: 0 4px 2px -2px gray;
  display: flex;
  justify-content: space-between;
  height: 130px;
  max-width: 350px;
  width: 100%;
  border: 0.7px solid rgba(220, 220, 220, 0.6);
  padding: 5%;
  margin-bottom: 15px;
  &:hover {
    box-shadow: none;
  }
`;

export default JobPosting;
