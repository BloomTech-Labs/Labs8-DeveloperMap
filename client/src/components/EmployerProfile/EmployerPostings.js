import React from 'react';
import firebase from '../../firebase/firebase';
import JobPosting from './JobPosting';
import FavHeart from '../../images/favorites-icon.png';
import Heart from '../../images/hollow-heart.png';
import axios from 'axios';

import { Posts, PostContainer } from './EmployerStyles';

class EmployerPostings extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  // favToggle = async (e, post) => {
  //   e.preventDefault();

  //   const user = this.props.user;
  //   if (!user) {
  //     this.props.history.push('/signin');
  //   } else if (e.target.src === FavHeart) {
  //     //Removes Favorited Post from current User if confirms
  //     const favoritedList = [
  //       ...this.state.favoritedList.filter(jobId => jobId !== e.target.id),
  //     ];
  //     this.setState({ favoritedList });
  //     e.target.src = Heart;
  //   } else {
  //     //Adds Favorited Post to current User's list if confirms
  //     const favoritedList = [...this.state.favoritedList];
  //     favoritedList.push(e.target.id);
  //     this.setState({ favoritedList });
  //     e.target.src = FavHeart;
  //   }
  // };

  render() {
    const { posts, favoritedList } = this.props;
    return (
      <PostContainer>
        {posts ? (
          <Posts>
            {Object.values(posts).map((post, i) => (
              <JobPosting
                key={i}
                index={i}
                favoritedList={favoritedList}
                {...this.props}
                post={post}
                favToggle={this.props.favToggle}
              />
            ))}
          </Posts>
        ) : (
          <Posts>
            <h1>This company has no job postings.</h1>
          </Posts>
        )}
      </PostContainer>
    );
  }
}

export default EmployerPostings;
