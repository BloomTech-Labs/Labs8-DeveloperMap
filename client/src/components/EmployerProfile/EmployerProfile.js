import React from 'react';
import firebase from '../../firebase/firebase';
import { ModalContainer, ModalMain } from '../../styles/ModalGlobalStyle';

import styled from 'styled-components';
import EmployerPostings from './EmployerPostings';

class EmployerProfile extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      companyName: '',
      companyWebsite: '',
      email: '',
      location: '',
      phoneNumber: '',
    };
  }

  componentDidMount() {
    const uid = this.props.match.params.companyId;
    firebase
      .database()
      .ref()
      .child(`companies/${uid}`)
      .once('value')
      .then(snapshot => {
        this.setState(snapshot.val());
      });
  }

  render() {
    return (
      <ModalContainer data-type="modal-container">
        <ModalMain>
          <Styling>
          <Info>
            <h2>{this.state.companyName}</h2>
            <h3>Location:</h3>
            <h4>{this.state.location}</h4>
            <h3>Website:</h3>
            <h4>{this.state.companyWebsite}</h4>
            <h3>Phone</h3>
            <h4>{this.state.phoneNumber}</h4>
            <h3>Email:</h3>
            <h4>{this.state.email}</h4>
          </Info>
          <EmployerPostings {...this.props}/>
          </Styling>
        </ModalMain>
      </ModalContainer>
    );
  }
  
}

const Info = styled.div`

  max-width: 400px;
  width: 100%;
`;


const Styling = styled.div`
  display: flex;
  padding: 5%;
  justify-content: space-between;
  max-width: 700px;
  width: 100%;
`;

export default EmployerProfile;
