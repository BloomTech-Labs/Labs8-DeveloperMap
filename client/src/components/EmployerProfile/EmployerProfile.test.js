import React from 'react';
import renderer from 'react-test-renderer';

import EmployerProfile from './EmployerProfile';

describe(`Employer Profile Component`, () => {
    it('renders correctly', () => {
        const tree = renderer.create(<EmployerProfile />).toJSON();
        expect(tree).toMatchSnapshot();
      });
})