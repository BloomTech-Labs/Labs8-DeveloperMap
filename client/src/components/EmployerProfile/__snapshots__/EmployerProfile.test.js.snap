// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`employer profile component should render without error 1`] = `
<div
  className="sc-bdVaJa sc-dnqmqq bWtbPz"
>
  <div
    className="sc-bwzfXH sc-iwsKbI GjyAO"
  >
    <div
      className="sc-gZMcBi kPhLgs"
    >
      <div
        className="globe-filter"
      />
      <div
        className="globe-spinning"
      />
    </div>
  </div>
</div>
`;
